dashboard/src/components/TimeSelect.tsx

```diff
@@ -0,0 +1,203 @@
+
+    handleUpdateButtonClick = (
+        event: React.SyntheticEvent<HTMLButtonElement>
+    ) => {
+        const newAutoUpdate = !(this.state as {autoUpdate?: boolean}).autoUpdate;
+        this.setState({autoUpdate: newAutoUpdate} as {}, () => {
+            const props = (this.props as Props);
+            if (props.onAutoUpdateChange) {
+                return props.onAutoUpdateChange(newAutoUpdate);
+            }
+        });
+    }
+
+    handleInputKey(event: React.KeyboardEvent<HTMLInputElement>) {
+        if (event.key === 'Enter') {
+            const {inputValue, inputFormat} = this.state;
+            // Parse the input value to moment object in non-strict mode
+            const time = moment(inputValue, inputFormat, false);
+            if (time.isValid()) {
+                this.setTimeValue(time);
+            }
+        }
+        ReactDatetimeProto.onInputKey.call(this, event);
+    }
+
+    setTimeValue(time: moment.Moment) {
+        const event = {target: {value: time.format(this.state.inputFormat)}};
+        ReactDatetimeProto.onInputChange.call(this, event);
+    }
+
+    render() {
+        let className = 'rdt' + ((this.state.open) ? ' rdtOpen' : '');
+        const icon = ((this.state as Props).autoUpdate) ? 'clock-o' : 'circle-o';
+        const iconClass = 'fa fa-' + icon;
+        const inputProps = {
+            key: 'i',
+            type: 'text' as 'text',
+            className: 'form-control',
+            onFocus: (this as ReactDatetimeInt).openCalendar,
+            onChange: (this as ReactDatetimeInt).onInputChange,
+            onKeyDown: this.handleInputKey.bind(this),
+            value: this.state.inputValue,
+        };
+
+        const getComponentProps = (this as {getComponentProps?: () => {}}).getComponentProps;
+        const componentProps = (getComponentProps) ? getComponentProps() : {};
+        const handleClickOutside = ReactDatetimeProto.handleClickOutside.bind(this);
+
+        return (
+            <div className={className}>
+                <InputGroup>
+                    <Input {...inputProps}/>
+                    <div key="dt" className="rdtPicker">
+                        <CalendarContainer
+                            view={(this.state as {currentView?: string}).currentView || ''}
+                            viewProps={componentProps}
+                            onClickOutside={handleClickOutside}
+                        />
+                    </div>
+                    <InputGroupButton>
+                        <Button
+                            onClick={this.handleUpdateButtonClick}
+                            className="update-button"
+                            color="info"
+                            outline={true}
+                            active={(this.state as {autoUpdate?: boolean}).autoUpdate}
+                        >
+                            <i className={iconClass}/>
+                            <span className="text"> Pidä ajan tasalla</span>
```

Comment:
Space before text, is that on purpose? If yes, consider using CSS.

Reply:
Yes it's on purpose. Don't know how to do that with CSS.

Reply:
Ok